### PR TITLE
refactor(curriculum): updating step 22 of stats calculator project

### PR DESCRIPTION
**Pull request by jdwilkin4**
## Summary of changes

This step comes up a lot on the forum. The issues seem to be the strictness of the tests and the understanding of the problem. The original step had a lot of the answer baked into the hints. But based on forum interaction, it looks like people are copying the answer from the hints without understanding what it means. So I moved the answers out of the hints and expanded the description instead. Also, I removed the separate step to return median and just moved that into step 22. 
That way are tests just check for the correct return value without a regex test.  This will allow for greater flexibility in correct answers from campers. 

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Not sure why the test for the Build new API container is failing 🤔

**Comment by Supravisor** [comment]
Hi @jdwilkin4 , may I work on this one?  I had similar issues with CI - Node.js / Test italian and portuguese

**Comment by jdwilkin4** [comment]
@Supravisor 

Any issues dealing with CI/CD stuff is handled by the dev team

**Comment by naomi-lgbt** [comment]
Sorry @jdwilkin4 I merged the step 27 PR and turned this into a mess.

**Comment by jdwilkin4** [comment]
@naomi-lgbt 

There might be a way to experiment with breaking this up into multiple steps.

I would have to play around with it.
But if broken up differently it would help with the confusion on the forum
https://forum.freecodecamp.org/search?q=step%2022%20%23javascript%20Statistics

Also, we could still test the final result so it allows for more flexibility in answers like this one
https://forum.freecodecamp.org/t/learn-advanced-array-methods-by-building-a-statistics-calculator-step-22/670107/2

**Comment by naomi-lgbt** [comment]
Yes, I'd definitely love to address the consistent tripping points on the forum.

But I also struggled with the state of step 22 here, myself 😅

**Comment by jdwilkin4** [comment]
@naomi-lgbt 

I broke up the step, into a few steps.

The first one focuses on how to check if array length is even or not. 
The second one focuses on how to find the median for odd length list of numbers
The third one focuses on how to find the median for even length list of numbers
The fourth one asks them to remove the test code

Then the last one asks them to put all they learned together to check if array is even. If so return the median for that, otherwise return the median for odd list. This lets campers write whatever solution they want to.

I removed the requirement for create the median variable and ternary.
The way I see it, if/else should be a viable solution.
As long as they return the right info, then it should be good.

I tested it out with if/else solution, a ternary solution, the original solution, and a solution created different variables for middle numbers and an if/else. They are passing on my end

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/661890c4abae9f2a0eddad6b.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/661890c4abae9f2a0eddad6b.md:12]:
```suggestion
To check if a number is even or odd, you can use the <dfn>modulus operator</dfn> `%`. The modulus operator returns the remainder of the division of two numbers. 
```
Non-blocking nit

**Review by naomi-lgbt — APPROVED** [review]

**Review by Ksound22 — APPROVED** [review]
Well done, @jdwilkin4!

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md:21]:
```suggestion
Your `getMedian` function should return the mean of the middle two numbers if the array length is even.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md:12]:
```suggestion
Inside your `getMedian` function, check if the length of `array` is even. If it is, find the middle two numbers, calculate their mean, and return the result. If the length of `array` is odd, return the middle number.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md:16]:
```suggestion
Also, if you need help, refer back to the previous steps to see how to find the median for an array.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/661899113600ef307d8a7487.md
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/661899113600ef307d8a7487.md:10]:
```suggestion
To get the median of an array with an odd number of elements, you will need to find and return the middle number.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/6618a3264f955339063d3328.md
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/6618a3264f955339063d3328.md:10]:
```suggestion
To find the median of an even list of numbers, you need to find the two middle numbers and calculate the mean of those numbers. 
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/6618abf6368d073f0f659780.md
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/6618abf6368d073f0f659780.md:10]:
```suggestion
Now that you have a better understanding of how to find the median for odd and even lists of numbers, you can remove all your test code from the previous steps.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Comment by Ksound22** [comment]
This has two approvals already. Why is not yet merged?

**Review by Sembauke — APPROVED** [review]

**Comment by jdwilkin4** [comment]
@Ksound22 

> This has two approvals already. Why is not yet merged?

When someone requests changes that need to be made, the person who requested them needs to approve those changes before it can be merged in. Otherwise, merging is blocked for the PR.

That is how the fcc repo is setup

**Pull request merged**
